app/actions/nodeStorageActions.js

```diff
@@ -1,17 +1,78 @@
 // @flow
 import { createActions } from 'spunky'
 import { isEmpty } from 'lodash-es'
+import { rpc } from 'neon-js'
 
 import { getStorage, setStorage } from '../core/storage'
+import {
+  MAIN_NETWORK_ID,
+  TEST_NETWORK_ID,
+  NODES_MAIN_NET,
+  NODES_TEST_NET,
+  NODE_EXLUSION_CRITERIA
+} from '../core/constants'
+import { findNetworkByLabel } from '../core/networks'
 
 const ID = 'nodeStorage'
 const STORAGE_KEY = 'selectedNode'
+let cachedRPCUrl
 
 type Props = {
   url: string,
   net: string
 }
 
+export const getRPCEndpoint = async (
+  net: string,
+  excludeCritera: Array<string> = NODE_EXLUSION_CRITERIA
+) => {
+  try {
+    if (cachedRPCUrl) return cachedRPCUrl
+    const NETWORK = findNetworkByLabel(net)
+    let nodeList
+    switch (NETWORK.id) {
+      case MAIN_NETWORK_ID:
+        nodeList = NODES_MAIN_NET
+        break
+      case TEST_NETWORK_ID:
+        nodeList = NODES_TEST_NET
+        break
+      default:
+        nodeList = NODES_MAIN_NET
+    }
+    const data = nodeList
+      .filter(
+        data => !excludeCritera.some(criteria => data.url.includes(criteria))
+      )
+      .map(data => {
+        let url = data.protocol ? `${data.protocol}://${data.url}` : data.url
+        url = data.port ? `${url}:${data.port}` : url
+        const client = new rpc.RPCClient(url)
+        // eslint-disable-next-line
+        data.client = client
+        return data
+      })
+    await Promise.all(data.map(data => data.client.ping()))
+    const nodes = data.sort(
+      (a, b) => b.client.lastSeenHeight - a.client.lastSeenHeight
+    )
+    if (nodes.length === 0) throw new Error('No eligible nodes found!')
+    const heightThreshold = nodes[0].client.lastSeenHeight - 1
+    const goodNodes = nodes.filter(
+      n => n.client.lastSeenHeight >= heightThreshold
+    )
+    const randomlySelectedRPCUrl =
+      goodNodes[Math.floor(Math.random() * goodNodes.length)].client.net
```

Comment:
I think its better to use lodash `_.random` - https://lodash.com/docs/4.17.11#random

Reply:
👍